### PR TITLE
add replicated cli to arkade get

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
## Description

This PR adds `replicated` cli to `arkade get`.

## Motivation and Context

- [x] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))
Fixes: #912 

## How Has This Been Tested?

Running the `arkade get replicated` locally 
```
> ./arkade get replicated
Downloading: replicated
2023/05/23 08:57:09 Looking up version for replicated
2023/05/23 08:57:09 Found: v0.45.0
Downloading: https://github.com/replicatedhq/replicated/releases/download/v0.45.0/replicated_0.45.0_linux_amd64.tar.gz
5.94 MiB / 5.94 MiB [----------------------------------------------------------------------------------------------] 100.00%
/tmp/replicated_0.45.0_linux_amd64.tar.gz written.
2023/05/23 08:57:12 Looking up version for replicated
2023/05/23 08:57:12 Found: v0.45.0
2023/05/23 08:57:12 Extracted: /tmp/replicated
2023/05/23 08:57:12 Copying /tmp/replicated to /home/nikhilsharmawe/.arkade/bin/replicated

Wrote: /home/nikhilsharmawe/.arkade/bin/replicated (16.06MB)

# Add arkade binary directory to your PATH variable
export PATH=$PATH:$HOME/.arkade/bin/

# Test the binary:
/home/nikhilsharmawe/.arkade/bin/replicated

# Or install with:
sudo mv /home/nikhilsharmawe/.arkade/bin/replicated /usr/local/bin/

🐳 arkade needs your support: https://github.com/sponsors/alexellis
```
**Demo**: https://asciinema.org/a/RDpIGyEg4EEA0HPVbEm5vPcwu

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [x] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: pkg/get/tools.go [hunk 1]
```diff
@@ -3487,5 +3487,32 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
...
+				{{- if eq .OS "darwin" -}}
+				{{$osStr = "darwin"}}
+				{{$arch = "all"}}
+				{{- else if eq .OS "linux" -}}
+				{{$osStr = "linux"}}
+				{{- end -}}
+
+				{{$extStr := "tar.gz"}}
+				{{$versionStr := .Version}}
+				{{$versionStr = ReplaceFirst $versionStr "v" "" 1}}
```
Comment (alexellis) [inline pkg/get/tools.go:3513]: Can you explain briefly for me why we needed ReplaceFirst for this CLI but haven't needed it anywhere else?

Comment (alexellis) [reply pkg/get/tools.go:3513]: Seems like

https://pkg.go.dev/strings#TrimPrefix

or

https://pkg.go.dev/strings#TrimLeft

Would have worked here too and been a bit more intuitive, if we're just taking off the "v"?

What did we do elsewhere? This pattern is not unique to this CLI.

File: pkg/get/tools.go [hunk 2]
```diff
@@ -3487,5 +3487,32 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
...
 		})
+
+	tools = append(tools,
+		Tool{
+			Owner:       "replicatedhq",
+			Repo:        "replicated",
+			Name:        "replicated",
+			Description: "CLI for interacting with the Replicated Vendor API",
+			URLTemplate: `
+				{{$arch := "386"}}
```
Comment (alexellis) [inline pkg/get/tools.go:3498]: FYI we don't support 386 in any other apps, could you remove support?

**Comment by alexellis** [comment]
We need to see the tests run for downloading each architecture / OS combination.

Can you check out previous PRs for an example?

https://github.com/alexellis/arkade/blob/master/hack/test-tool.sh

File: pkg/get/tools.go (hunk 2 above)
Comment (NikhilSharmaWe) [reply pkg/get/tools.go:3498]: We support in the case of `helmfile`: https://github.com/alexellis/arkade/blob/d8986d8582d4706e986c3e7695cdc69abc15120b/pkg/get/tools.go#L88

Comment (NikhilSharmaWe) [reply pkg/get/tools.go]: Removed it.

But replicated-cli has a distribution for arch `386`: https://github.com/replicatedhq/replicated/releases/download/v0.45.0/replicated_0.45.0_linux_386.tar.gz

and in helmfile we add arch `386`: https://github.com/alexellis/arkade/blob/master/pkg/get/tools.go#L88

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Thanks for the :tada: @roderik. Is this a CLI that you needed?

**Pull request merged**